public/app/features/auth-config/components/ServerDiscoveryModal.tsx

```diff
@@ -0,0 +1,78 @@
+import { JSX } from 'react';
+import { useForm } from 'react-hook-form';
+import { connect, ConnectedProps } from 'react-redux';
+
+import { Button, Input, Field, Modal } from '@grafana/ui';
+
+import { Trans } from '../../../core/internationalization';
+import { ServerDiscoveryFormData } from '../types';
+import { isUrlValid } from '../utils/url';
+
+interface OwnProps {
+  isOpen: boolean | undefined;
+  onClose: () => void;
+  onSuccess: (data: ServerDiscoveryFormData) => void;
+}
+
+export type Props = OwnProps & ConnectedProps<typeof connector>;
+
+const connector = connect(undefined, {});
+
+export const ServerDiscoveryModalUnconnected = ({ isOpen, onClose, onSuccess }: Props): JSX.Element => {
```

Comment:
```suggestion
export const ServerDiscoveryModalUnconnected = ({ isOpen, onClose, onSuccess }: Props) => {
```
The return type is inferred by TS here.

Reply:
Done.